ode-wrapper/src/main/java/com/trihydro/odewrapper/controller/WydotTimRcController.java

```diff
@@ -60,29 +63,34 @@ public ResponseEntity<String> createUpdateRoadConditionsTim(@RequestBody TimRcLi
 
         utility.logWithDate(dateFormat.format(date) + " - Create Update RC TIM", this.getClass());
         String post = gson.toJson(timRcList);
-        utility.logWithDate(post.toString(), this.getClass());
+        utility.logWithDate(post, this.getClass());
 
-        List<ControllerResult> resultList = new ArrayList<ControllerResult>();
-        List<ControllerResult> errList = new ArrayList<ControllerResult>();
-        ControllerResult resultTim = null;
-        List<WydotTim> timsToSend = new ArrayList<WydotTim>();
+        List<ControllerResult> resultList = new ArrayList<>();
+        List<ControllerResult> errList = new ArrayList<>();
+        ControllerResult resultTim;
+        List<WydotTim> timsToSend = new ArrayList<>();
 
         // build TIM
         for (WydotTimRc wydotTim : timRcList.getTimRcList()) {
 
             resultTim = validateInputRc(wydotTim);
 
-            // workaround for geometry start/end point population 
-            WydotTimRc wydotTimRc = wydotTim.copy();
-
-            if (resultTim.getResultMessages().size() > 0) {
+            if (!resultTim.getResultMessages().isEmpty()) {
                 resultList.add(resultTim);
                 errList.add(resultTim);
                 continue;
             }
 
-            // add TIM to list for processing later
-            timsToSend.add(wydotTimRc);
+            // Each ITIS string in the TIM corresponds to a single TIM to be sent to ODE
+            for (String itisCodeEntry : wydotTim.getItisCodes()) {

```

Comment:
Question (blocking): Is accepting ITIS codes directly in this controller compatible with the CTW standards, or should we move to a human-readable label?